tests/integration/cartography/intel/aws/ec2/test_ec2_instances.py

```diff
@@ -90,3 +93,87 @@ def test_ec2_reservations_to_instances(neo4j_session, *args):
         for n in nodes
     }
     assert actual_nodes == expected_nodes
+
+
+def test_ec2_iaminstanceprofiles(neo4j_session, *args):
+    """
+    Ensure that iaminstanceprofiles are connected to their expected iam roles
+    """
+    neo4j_session.run(
+        """
+        MERGE (aws:AWSAccount{id: {aws_account_id}})
+        ON CREATE SET aws.firstseen = timestamp()
+        SET aws.lastupdated = {aws_update_tag}
+        """,
+        aws_account_id="OWNER_ACCOUNT_ID",
+        aws_update_tag=TEST_UPDATE_TAG,
+    )
+    neo4j_session.run(
+        """
+        MERGE (aws:AWSAccount{id: {aws_account_id}})
+        ON CREATE SET aws.firstseen = timestamp()
+        SET aws.lastupdated = {aws_update_tag}
+        """,
+        aws_account_id="ACCOUNT_NUM",
+        aws_update_tag=TEST_UPDATE_TAG,
+    )
+    
+    data_instances = tests.data.aws.ec2.instances.DESCRIBE_INSTANCES['Reservations']
+    data_iam = tests.data.aws.ec2.instances.DESCRIBE_INSTANCES['Roles']
+
+    cartography.intel.aws.ec2.instances.load_ec2_instances(
+        neo4j_session, data_instances, TEST_REGION, TEST_ACCOUNT_ID, TEST_UPDATE_TAG,
+    )
+
+    cartography.intel.aws.iam.load_roles(
+        neo4j_session, data_iam, TEST_ACCOUNT_ID, TEST_UPDATE_TAG,
+    )
+
+    config = cartography.config.Config(
+        neo4j_uri='bolt://localhost:7687',
+        update_tag=TEST_UPDATE_TAG,
+        aws_sync_all_profiles=True,
+    )
+
+    common_job_parameters = {
+        "UPDATE_TAG": config.update_tag,
+        "permission_relationships_file": config.permission_relationships_file,
```

Comment:
Remove `'permission_relationships_file'`; this field is not necessary for this test.